### ISSUE TITLE
Log-write safety does not cover version-control commands, and a git-tracked log is exactly where they get run

### ISSUE THREAD

**Issue by thanhvu-create** [rebelytics/one-skill-to-rule-them-all#36]
First, thanks for the skill — the Log-write safety section has already paid for itself here, and the check-then-act-then-verify numbering discipline caught a real parallel-session collision.

This is a gap in that same section rather than a bug.

## What happened

The observation log lives in a git repository (the whole agent-config directory is versioned, so the log is versioned along with it).

A session needed to switch branches to sync with a merged PR. The working tree was dirty — the dirt was the log, which had an observation appended minutes earlier. To unblock the branch switch, the session ran:

```
git checkout -- skill-observations/log.md
```

That reverted the new observation. Meanwhile a concurrent session had already read the log, seen that observation, and appended three more — one of which cross-referenced the reverted entry by number. The result was a log whose entries pointed at a number that no longer existed.

It was recoverable only because the entry text happened to still be in the reverting session's context. A session that had appended and moved on would have lost it silently, and the rule-5 survival check at session end would have correctly reported the loss with no way to repair it.

## Why the existing rules don't catch this

The Log-write safety section is thorough about *programmatic* mutation — archival, renumbering, status edits, full-file rewrites — and mandates backup, re-read-and-merge, bounded per-entry edits, structural invariants, and survival checks.

Version-control commands mutate the file just as destructively, but:

1. **They don't read as edits.** `git checkout --`, `git stash`, `git reset --hard`, a branch switch carrying a local modification, a rebase that drops a hunk — none of these look like "writing to the log", so none of them trip the section's discipline.
2. **They get run reflexively, as housekeeping.** The goal in the moment is "make the tree clean enough to switch branches", not "modify the log". The log is collateral.
3. **A continuously-appended log is almost always dirty**, which is what makes the reflexive revert attractive in the first place. This compounds with any dirty-tree precondition elsewhere in a workflow: a guard that demands a clean tree, applied to a repo whose log is always dirty, actively teaches sessions to clear it.

## Suggested change

Add a rule 6 to Log-write safety, roughly:

> **Treat version-control commands as mutations when the log is tracked.** Rules 1–5 cover programmatic edits. A log kept in a git repository is destroyed just as thoroughly by `git checkout -- <log>`, `git stash`, `git reset --hard`, a branch switch carrying a local modification, or a rebase that drops a hunk. Before any git operation that can discard working-tree state, copy the log somewhere outside the repository; afterwards, re-run the rule-5 survival check and re-append from the copy if anything is missing. **Prefer committing pending entries over reverting them** — when the dirt in the tree is the log itself, a commit is always the cheaper way to get clean. Scope any dirty-tree guard to exclude the log rather than teaching sessions to clear it.

And a Quick Reference row:

> \| Log is git-tracked and a git command wants to discard it? \| Same discipline — `checkout --`/`stash`/`reset --hard`/branch switch are mutations. Copy outside the repo first, prefer committing over reverting, re-run the survival check after \|

## The generalisable principle

A safety discipline scoped to one class of mutation leaves every other class unguarded, and the unguarded classes are the dangerous ones precisely because they do not look like edits. Once shared state is placed under version control, the version-control commands become mutation operations and inherit every rule written for the mutations they resemble.

This may also be worth a line in the environments reference: versioning the workspace folder is good practice for exactly the rollback reasons the skill cares about, and it introduces this new mutation surface at the same time.

---

*Filed from real usage on a Windows/Claude Code setup with two concurrent sessions writing one pinned global log.*

**Comment by rebelytics** [comment]
Shipped in v3.0.0, released today: a 'workspace under version control' section treats git commands as mutations, names git clean as the biggest threat to untracked new observations, and prefers committing over reverting. You're credited as co-author on the commit. Thanks again for your contribution.
